### ISSUE TITLE
false positive on use strict directive

### ISSUE THREAD

**Issue by franck-paul** [biomejs/biome#3524]
### Environment information

```bash
Sublime Text 4 (last version), lsp-biome package installed (1.8.2)
My biome.json:

{
  "organizeImports": { "enabled": true },
  "linter": {
    "enabled": true,
    "rules": {
      "recommended": true
    }
  }
}

The JS source file (not a module) opened (in ST):

/*global dotclear */
'use strict';

dotclear.ready(() => {
  // DOM ready and content loaded

  // Give focus to user field
  /**
   * @type {HTMLInputElement|null}
   */
  const uid = document.querySelector('input[name=user_id]');
  if (uid) uid?.focus();

  /**
   * @type {HTMLElement|null}
   */
  const ckh = document.getElementById('cookie_help');
  if (ckh) ckh.style.display = navigator.cookieEnabled ? 'none' : '';

  /**
   * @type {HTMLInputElement|null}
   */
  const upw = document.querySelector('input[name=user_pwd]');
  if (!upw || !uid) {
    return;
  }

  // Add an event listener to capture Enter key press in user field to give to password field if it is empty
  uid.addEventListener('keypress', (/** @type {KeyboardEvent} */ event) => {
    if (event.key === 'Enter' && upw.value === '') {
      // Password is empty, give focus to it
      upw.focus();
      // Stop handling of this event (Enter key pressed)
      event.preventDefault();
    }
  });
});
```

### Rule name

noRedundantUseStrict

### Playground link

https://biomejs.dev/playground/?code=LwAqAGcAbABvAGIAYQBsACAAZABvAHQAYwBsAGUAYQByACAAKgAvAAoAJwB1AHMAZQAgAHMAdAByAGkAYwB0ACcAOwAKAAoAZABvAHQAYwBsAGUAYQByAC4AcgBlAGEAZAB5ACgAKAApACAAPQA%2BACAAewAKACAAIAAvAC8AIABEAE8ATQAgAHIAZQBhAGQAeQAgAGEAbgBkACAAYwBvAG4AdABlAG4AdAAgAGwAbwBhAGQAZQBkAAoACgAgACAALwAvACAARwBpAHYAZQAgAGYAbwBjAHUAcwAgAHQAbwAgAHUAcwBlAHIAIABmAGkAZQBsAGQACgAgACAALwAqACoACgAgACAAIAAqACAAQAB0AHkAcABlACAAewBIAFQATQBMAEkAbgBwAHUAdABFAGwAZQBtAGUAbgB0AHwAbgB1AGwAbAB9AAoAIAAgACAAKgAvAAoAIAAgAGMAbwBuAHMAdAAgAHUAaQBkACAAPQAgAGQAbwBjAHUAbQBlAG4AdAAuAHEAdQBlAHIAeQBTAGUAbABlAGMAdABvAHIAKAAnAGkAbgBwAHUAdABbAG4AYQBtAGUAPQB1AHMAZQByAF8AaQBkAF0AJwApADsACgAgACAAaQBmACAAKAB1AGkAZAApACAAdQBpAGQAPwAuAGYAbwBjAHUAcwAoACkAOwAKAAoAIAAgAC8AKgAqAAoAIAAgACAAKgAgAEAAdAB5AHAAZQAgAHsASABUAE0ATABFAGwAZQBtAGUAbgB0AHwAbgB1AGwAbAB9AAoAIAAgACAAKgAvAAoAIAAgAGMAbwBuAHMAdAAgAGMAawBoACAAPQAgAGQAbwBjAHUAbQBlAG4AdAAuAGcAZQB0AEUAbABlAG0AZQBuAHQAQgB5AEkAZAAoACcAYwBvAG8AawBpAGUAXwBoAGUAbABwACcAKQA7AAoAIAAgAGkAZgAgACgAYwBrAGgAKQAgAGMAawBoAC4AcwB0AHkAbABlAC4AZABpAHMAcABsAGEAeQAgAD0AIABuAGEAdgBpAGcAYQB0AG8AcgAuAGMAbwBvAGsAaQBlAEUAbgBhAGIAbABlAGQAIAA%2FACAAJwBuAG8AbgBlACcAIAA6ACAAJwAnADsACgAKACAAIAAvACoAKgAKACAAIAAgACoAIABAAHQAeQBwAGUAIAB7AEgAVABNAEwASQBuAHAAdQB0AEUAbABlAG0AZQBuAHQAfABuAHUAbABsAH0ACgAgACAAIAAqAC8ACgAgACAAYwBvAG4AcwB0ACAAdQBwAHcAIAA9ACAAZABvAGMAdQBtAGUAbgB0AC4AcQB1AGUAcgB5AFMAZQBsAGUAYwB0AG8AcgAoACcAaQBuAHAAdQB0AFsAbgBhAG0AZQA9AHUAcwBlAHIAXwBwAHcAZABdACcAKQA7AAoAIAAgAGkAZgAgACgAIQB1AHAAdwAgAHwAfAAgACEAdQBpAGQAKQAgAHsACgAgACAAIAAgAHIAZQB0AHUAcgBuADsACgAgACAAfQAKAAoAIAAgAC8ALwAgAEEAZABkACAAYQBuACAAZQB2AGUAbgB0ACAAbABpAHMAdABlAG4AZQByACAAdABvACAAYwBhAHAAdAB1AHIAZQAgAEUAbgB0AGUAcgAgAGsAZQB5ACAAcAByAGUAcwBzACAAaQBuACAAdQBzAGUAcgAgAGYAaQBlAGwAZAAgAHQAbwAgAGcAaQB2AGUAIAB0AG8AIABwAGEAcwBzAHcAbwByAGQAIABmAGkAZQBsAGQAIABpAGYAIABpAHQAIABpAHMAIABlAG0AcAB0AHkACgAgACAAdQBpAGQALgBhAGQAZABFAHYAZQBuAHQATABpAHMAdABlAG4AZQByACgAJwBrAGUAeQBwAHIAZQBzAHMAJwAsACAAKAAvACoAKgAgAEAAdAB5AHAAZQAgAHsASwBlAHkAYgBvAGEAcgBkAEUAdgBlAG4AdAB9ACAAKgAvACAAZQB2AGUAbgB0ACkAIAA9AD4AIAB7AAoAIAAgACAAIABpAGYAIAAoAGUAdgBlAG4AdAAuAGsAZQB5ACAAPQA9AD0AIAAnAEUAbgB0AGUAcgAnACAAJgAmACAAdQBwAHcALgB2AGEAbAB1AGUAIAA9AD0APQAgACcAJwApACAAewAKACAAIAAgACAAIAAgAC8ALwAgAFAAYQBzAHMAdwBvAHIAZAAgAGkAcwAgAGUAbQBwAHQAeQAsACAAZwBpAHYAZQAgAGYAbwBjAHUAcwAgAHQAbwAgAGkAdAAKACAAIAAgACAAIAAgAHUAcAB3AC4AZgBvAGMAdQBzACgAKQA7AAoAIAAgACAAIAAgACAALwAvACAAUwB0AG8AcAAgAGgAYQBuAGQAbABpAG4AZwAgAG8AZgAgAHQAaABpAHMAIABlAHYAZQBuAHQAIAAoAEUAbgB0AGUAcgAgAGsAZQB5ACAAcAByAGUAcwBzAGUAZAApAAoAIAAgACAAIAAgACAAZQB2AGUAbgB0AC4AcAByAGUAdgBlAG4AdABEAGUAZgBhAHUAbAB0ACgAKQA7AAoAIAAgACAAIAB9AAoAIAAgAH0AKQA7AAoAfQApADsACgA%3D

### Expected result

It should not thrown an error on `'use strict';` line

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
What is the extension of your file? Have you set `package.json`' `type` field to `commonjs`?

**Comment by franck-paul** [comment]
My files are all .js
And I have no package.json, the JS files are part of a blog application (mainly written in PHP).

**Comment by ematipico** [comment]
By default, Biome treats `.js` files as **modules**. We take the assumptions that the majority of the JS files are part of modern projects. That's why it emits that "false positive".

I think, for now, the better course of action is to disable the rule or suppress it. We will think about ways to change this via programmatic configuration.

**Comment by franck-paul** [comment]
Ok, the rule is already disabled, thanks for your time about this issue!
